src/Number.php

```diff
@@ -20,17 +20,37 @@
     private $fractionalPart;
 
     /**
+     * @var array
+     */
+    private static $numbers = [0 => 1, 1 => 1, 2 => 1, 3 => 1, 4 => 1, 5 => 1, 6 => 1, 7 => 1, 8 => 1, 9 => 1];
+
+    /**
+     * @var array
+     */
+    private static $signs = ['-' => 1];
+
+    /**
      * @param string $integerPart
      * @param string $fractionalPart
      */
     public function __construct($integerPart, $fractionalPart = '')
     {
-        if ($this->validateNumberAsInteger($integerPart) === false) {
-            throw new \InvalidArgumentException('Invalid number');
+        if ($integerPart !== '' && $this->validateAsIntegerPart($integerPart) === false) {
+            throw new \InvalidArgumentException(
+                'Invalid number, integer part '.$integerPart.' is not an integer'
+            );
         }
 
-        if ($fractionalPart !== '' && $this->validateNumberAsInteger($fractionalPart) === false) {
-            throw new \InvalidArgumentException('Invalid number');
+        if ($fractionalPart !== '' && $this->validateAsFractionalPart($fractionalPart) === false) {
+            throw new \InvalidArgumentException(
+                'Invalid number, fractional part '.$integerPart.' is not an integer'
+            );
+        }
+
+        if ($integerPart === '-') {
```

Comment:
We are supporting `.5` in the current version due to our current number validation. With `.5` the integer part is empty, str_split results in an empty array, so `count($invalid) === 0` becomes true. Hence, we support `.5`.

So, because I do not want BC break, I continue supporting this. But then we have to be consequent. This means `.5` and `-.5` are now supported. Hence, `-` is a valid integer part and should result in `-0`. We might want to undo this in release where we do BC again.